[F11-2] Concierge

adamwth opened the pull request:
@pikulet 
@teowz46 
@JiaqingTan 
@neilish3re

liliwei25 reviewed (commented):
Proceed to work towards v1.1. 👍

damithc commented:
Guys, the current version of your team repo is failing CI checks. Try to fix before v1.1

liliwei25 commented on LICENSE:
```diff
@@ -2,11 +2,11 @@ MIT License
 Copyright (c) 2016 Software Engineering Education - FOSS Resources
 
-Permission is hereby granted, free of charge, to any person obtaining a copy
+Permission is hereby granted, free of charge, to any guest obtaining a copy
```
I don't think you're supposed to edit the license

adamwth replied:
Oops, thanks!

liliwei25 commented on docs/UserGuide.adoc:
```diff
@@ -56,103 +58,214 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+Lists all the available rooms of that type for the next NUM_DAYS days +
+Format: `avail r/ROOM TYPE num/NUM_DAYS`
+
+Lists all available rooms from start to end date. Dates have to be given in YYMMDD format
```
This is confusing. Is this searched from a start to end date or is this searched for next NUM_DAYS? And where is the YYMMDD date used as input?

liliwei25 commented on docs/UserGuide.adoc:
```diff
@@ -56,103 +58,214 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
-=== Adding a person: `add`
 
-Adds a person to the address book +
+=== Clearing all entries : `avail`
```
is 'avail' used for clearing all entries?

liliwei25 commented on docs/UserGuide.adoc:
```diff
@@ -56,103 +58,214 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+
+Examples:
+
+* `avail r/Double Room num/5`
```
These examples for 'avail' is separated from the 'avail' command description by the 'add' command

liliwei25 commented on docs/UserGuide.adoc:
```diff
@@ -56,103 +58,214 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+* `avail r/Double Room num/5`
+* `avail r/Double Room ds/20180910 de/21080912`
+
+=== Listing all guests : `list`
```
So are you listing all guests or all room types?

liliwei25 commented on docs/UserGuide.adoc:
```diff
@@ -56,103 +58,214 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+=== Mark Room as faulty: `fault`
+
+Marks the given room for maintenance +
+Format: `service rm/709 mt/wiring`
```
'fault' or 'service'?

liliwei25 commented on docs/templates/LICENSE:
```diff
@@ -5,11 +5,11 @@ MIT License
 Copyright (C) 2012-2018 Dan Allen, Ryan Waldron and the Asciidoctor Project
 
-Permission is hereby granted, free of charge, to any person obtaining a copy
+Permission is hereby granted, free of charge, to any guest obtaining a copy
```
Please don't change the licensing

liliwei25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -582,9 +582,9 @@ Do take a look at <<Design-Model>> before attempting to modify the `Model` compo
 ** The link:{repoURL}/src/main/java/seedu/address/model/Model.java[`Model`] and the link:{repoURL}/src/main/java/seedu/address/model/AddressBook.java[`AddressBook`] API need to be updated.
 ** Think about how you can use SLAP to design the method. Where should we place the main logic of deleting tags?
-**  Find out which of the existing API methods in  link:{repoURL}/src/main/java/seedu/address/model/AddressBook.java[`AddressBook`] and link:{repoURL}/src/main/java/seedu/address/model/person/Person.java[`Person`] classes can be used to implement the tag removal logic. link:{repoURL}/src/main/java/seedu/address/model/AddressBook.java[`AddressBook`] allows you to update a person, and link:{repoURL ...
+**  Find out which of the existing API methods in  link:{repoURL}/src/main/java/seedu/address/model/AddressBook.java[`AddressBook`] and link:{repoURL}/src/main/java/seedu/address/model/guest/Person.java[`Person`] classes can be used to implement the tag removal logic. link:{repoURL}/src/main/java/seedu/address/model/AddressBook.java[`AddressBook`] allows you to update a guest, and link:{repoURL}/ ...
```
/guest/Person.java or /person/Guest.java?

liliwei25 commented on src/main/java/seedu/address/logic/commands/AssignCommand.java:
```diff
@@ -0,0 +1,68 @@
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Assigns a guest to a room.\n"
+            + "Parameters: GUEST_NAME ROOM_NUMBER\n"
+            + "Example: " + COMMAND_WORD + " Thomas 056";
```
This format is different from the one specified in your User docs. Please make it consistent

liliwei25 commented on src/main/java/seedu/address/logic/commands/AssignCommand.java:
```diff
@@ -0,0 +1,68 @@
+        Guest guestToAssign = lastShownList.get(targetIndex.getZeroBased());
+        model.commitAddressBook();
+
+        return new CommandResult(String.format(MESSAGE_ASSIGN_GUEST_SUCCESS, guestToAssign));
```
String accepts two inputs but only one was given

liliwei25 commented on src/main/java/seedu/address/model/room/Capacity.java:
```diff
@@ -0,0 +1,51 @@
+    /**
+     * Returns true if a given int is a valid capacity.
+     */
+    public static boolean isValidCapacity(int test) {
```
'test' is not a good variable/parameter name

liliwei25 commented on src/test/java/seedu/address/logic/commands/CheckoutCommandTest.java:
```diff
@@ -0,0 +1,47 @@
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs(), getTypicalUniqueRoomList());
+    private CommandHistory commandHistory = new CommandHistory();
+
+    @Test
```
All the methods are empty

liliwei25 reviewed (commented):
There seems to some discrepancies in your docs. Please double check. 

Great effort! 👍

liliwei25 commented on commands_log/commands_log.txt:
```diff
@@ -0,0 +1,42 @@
+19/10/2018 06:33:22  a
```
Should not add this file to git

liliwei25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -319,6 +319,145 @@ image::UndoRedoActivityDiagram.png[width="650"]
+. `Room#checkout` will
+.. throw `NoBookingException` if the room has no bookings
+.. throw `NoActiveOrExpiredBookingException` if the room has no active or expired bookings
+.. delete the first booking from the list of bookings
```
Does it delete the first booking or booking of room "001"?

liliwei25 commented on src/main/java/seedu/address/logic/CommandHistory.java:
```diff
@@ -25,6 +26,22 @@ public CommandHistory(CommandHistory commandHistory) {
+            string.append(it.next() + newLine);
+        }
+        String inputString = string.toString();
+        System.out.println(inputString);
```
extra println statement?

liliwei25 commented on src/main/java/seedu/address/logic/commands/AddCommand.java:
```diff
@@ -20,52 +28,74 @@
         requireNonNull(model);
 
-        if (model.hasPerson(toAdd)) {
+        if (model.hasPerson(guestToAdd)) {
```
maybe can change method name to "hasGuest" to standardize

liliwei25 commented on src/main/java/seedu/address/logic/commands/AddCommand.java:
```diff
@@ -20,52 +28,74 @@
+        if (model.hasPerson(guestToAdd)) {
             throw new CommandException(MESSAGE_DUPLICATE_PERSON);
         }
+        model.addPerson(guestToAdd);
```
"addPerson" -> "addGuest"?

liliwei25 commented on src/main/java/seedu/address/logic/commands/CheckoutCommand.java:
```diff
@@ -30,10 +34,15 @@ public CheckoutCommand(RoomNumber roomNumber) {
+        }
         model.checkoutRoom(roomNumber);
-        model.commitRoomList();
+        model.commitAddressBook();
```
maybe can rename this method since this is not an addressbook anymore

liliwei25 commented on src/main/java/seedu/address/model/AddressBook.java:
```diff
@@ -15,7 +21,8 @@
 public class AddressBook implements ReadOnlyAddressBook {
```
maybe can rename the class as well since this is no longer an addressbook

liliwei25 commented on src/test/java/seedu/address/testutil/TypicalPersons.java:
```diff
@@ -15,62 +15,60 @@
 /**
  * A utility class containing a list of {@code Guest} objects to be used in tests.
  */
 public class TypicalPersons {
```
Maybe can rename to TypicalGuests?

liliwei25 commented on src/test/java/seedu/address/testutil/TypicalAddressBook.java:
```diff
@@ -0,0 +1,26 @@
+/**
+ * A utility class containing a {@code UniqueGuestList} and {@code UniqueRoomList} objects to be used in tests.
+ */
+public class TypicalAddressBook {
```
maybe can rename to something that suits your app?

liliwei25 commented on src/test/java/seedu/address/testutil/ExpenseBuilder.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    /**
+     * Initializes the ExpenseBuilder with the data of {@code bookingToCopy}.
```
typo

liliwei25 commented on src/main/java/seedu/address/model/room/booking/Booking.java:
```diff
@@ -0,0 +1,197 @@
+    // Identity fields
+    private final Guest guest;
+    private final BookingPeriod bookingPeriod;
+    private final Boolean checkIn;
```
check naming convention for booleans

liliwei25 reviewed (commented):
Good amount of effort! 👍 
* Some classes/methods/variables can be renamed since this is not an addressbook anymore
* The class diagrams should include the visibility of the attributes
* Should set up for reposense too

Commit:
Resolve merge conflict from #153 and #162

pikulet committed:
LogInCommandParser.java: Add parseAndHash method for password

pikulet committed:
ModelManager.java: Create username field

pikulet committed:
Model.java: Introduce InvalidPasswordException

Commit:
fix style

pikulet committed:
PasswordHashUtil.java: Add reference site

pikulet committed:
PasswordsStorage.java: Add password storage interface

pikulet committed:
UserPrefs.java: Add functionality to read in password file path

Commit:
XmlUtilTest and XmlSerializableConciergeTest: update with new format of valid concierge, refactor to remove SOME hard-coded test cases from AB4

Commit:
XmlSerializableConciergeTest: update

Commit:
TypicalConcerge: Refactor name of getTypicalConcierge -> getTypicalConciergeClean, and getTypicalConciergeWithRoomBookings -> getTypicalConcierge

Commit:
fix style

Commit:
XmlConciergeStorageTest: update tests and relevant xml files used for tests

Commit:
XmlSerializableConciergeTest: update tests and add relevant XML files for updated tests

pikulet committed:
Fix checkstyle errors

pikulet committed:
Merge branch 'master' into sign-in-feature

Commit:
XmlUtilTest: update the format of how tests are conducted, RoomListPanelHandle: fix NodeNotFoundException bug by changing FXML id of RoomCard

Commit:
DarkTheme.css: update with roomCardPane

pikulet committed:
JsonUtil.java: Add functionality to parse string to jsonnode

Commit:
fix codacy style

Commit:
fix style

adamwth committed:
[v1.3] XML tests (#166)

* BookingPeriod.java: update isValidBookingPeriod to test if start date is strictly before end date

* BookingTest and BookingPeriodTest: add

* BookingsTest: add, add compareTo method to Guest and update compareTo of BookingPeriod and Booking

* CheckinCommand: minor fix to check-in logic and refactor, Room and Capacity: refactor Capacity to be enum and delete SingleRoom, DoubleRoom, SuiteRoom. RoomCard and RoomDetailedCard: minor fix to FXML element naming and standardize description

* UniqueRoomList: refactor to enforce no missing rooms, other tests: update to reflect compulsory rooms in all concierge xml files

* CheckinCommand: change input format, add tokenizer for room prefix

* CheckInCommand: rename from CheckinCommand, .gitignore: update

* ConciergeBuilder: add withMenu

* ConciergeTest: update tests

* ConciergeTest, ModelManagerTest, CheckInCommandTest: add

* CheckinCommandTest: update

* test: update AddCommandSystemTest to actually add booking

* TypicalRooms: add support for preset rooms with bookings

* Test utils: add support for more reusable test cases in Typical<Class>.java, update CheckinCommandTest and CheckoutCommandTest to use added test utils

* XmlAdaptedBookingPeriodTest: add test

* XmlAdaptedBookingTest: add test

* XmlAdaptedRoomTest: add tests

* fix style

* XmlUtilTest and XmlSerializableConciergeTest: update with new format of valid concierge, refactor to remove SOME hard-coded test cases from AB4

* XmlSerializableConciergeTest: update

* TypicalConcerge: Refactor name of getTypicalConcierge -> getTypicalConciergeClean, and getTypicalConciergeWithRoomBookings -> getTypicalConcierge

* fix style

* XmlConciergeStorageTest: update tests and relevant xml files used for tests

* XmlSerializableConciergeTest: update tests and add relevant XML files for updated tests

* XmlUtilTest: update the format of how tests are conducted, RoomListPanelHandle: fix NodeNotFoundException bug by changing FXML id of RoomCard

* DarkTheme.css: update with roomCardPane

* fix codacy style

pikulet committed:
HashingException.java: move to model/util, Rename InvalidPasswordException to InvalidLogInException

pikulet committed:
InvalidLogInException: Add support for invalid username and password

pikulet committed:
Fix checkstyle errors

pikulet committed:
Merge branch 'master' into sign-in-feature

pikulet committed:
PasswordHashList.java: Add support for retrieving password hashes

pikulet committed:
LogInHelper.java: Add functionality to manage login

pikulet committed:
Model.java: redirect login features to LogInHelper

pikulet committed:
LogInCommand.java: Catch InvalidLogInException

pikulet committed:
Add Javadocs and fix checkstyle errors

adamwth committed:
Merge pull request #171 from adamwth/guest-checkin

[v1.3] Add CheckInCommandParserTest

pikulet committed:
PasswordHashList.java: Resolve NullPointerException caused by invalid username

pikulet committed:
ConciergeParser.java: Add functionality to accept login command

pikulet committed:
LogInHelper.java: Enforce lowercase comparison of hashed passwords

Commit:
Concierge.java: add support for checked-in guest list, Rooms and other classes: refactor

pikulet committed:
StorageManager.java: Add passwordstorage as parameter for constructor

pikulet committed:
LogInHelper.java: improve code quality with equalsIgnoreCase method

pikulet committed:
Fix checkstyle issues

pikulet committed:
Update data/passwords.json

Everyone's password is <user>-pw, for example:

user: pikulet
password: pikulet-pw

pikulet committed:
LogInCommand.java: Add case-sensitive specification

pikulet committed:
Merge branch 'master' into user-guide-update

pikulet committed:
Add login command documentation

pikulet committed:
Remove delete documentation, update command summary

pikulet committed:
Rename LogInHelper to LogInExecutor

pikulet committed:
Add LogInVerifier.java and UnauthorisedCommandException

pikulet committed:
LogicManager.java: Add sign-in verification

pikulet committed:
Model.java: add method to get username

pikulet committed:
Rename LogInExecutor to LogInManager

pikulet committed:
AddCommandTest.java: Override "getUsername" method

pikulet committed:
Add LogInUtil to allow for simulated log-ins

pikulet committed:
SystemTests: Add login before attempting to execute delete

pikulet committed:
pass_testing.json: change password to hashed version

pikulet committed:
Fix checkstyle errors

pikulet committed:
ClearCommandSystemTest.java: remove support for asserting login success

pikulet committed:
Fix checkstyle issues (for travis)

Commit:
TypicalConcierge & TypicalRooms: Refactor to move all typical room settings to TypicalConcierge

teowz46 committed:
all: fixed some Codacy detected issues

Only did a few, generally those related to my stuff.

teowz46 committed:
XmlSerializableConcierge: change menu storage to list

Currently, the menu is being stored in the XML storage file in the form of a HashMap<String, XmlAdaptedExpenseType>, where the key is the item number and the value is the corresponding expense type.

There is data redundancy as the item number appears in both the key and value. In addition, undesirable things may happen if the item number in the key is different from the item number in the value.

Such coupling has been removed, by storing the menu as a list instead. Thus, the item number only appears once for every entry, and every item number can be assumed to be correct and independent of other item numbers. In addition, additional checks were added to ensure that the menu does not contain items with duplicate item numbers.

teowz46 committed:
Merge remote-tracking branch 'upstream/master' into service

# Conflicts:
#	src/main/java/seedu/address/logic/parser/CliSyntax.java
#	src/main/java/seedu/address/model/Model.java
#	src/main/java/seedu/address/model/ModelManager.java
#	src/main/java/seedu/address/model/room/Room.java
#	src/main/java/seedu/address/storage/XmlSerializableConcierge.java
#	src/test/java/seedu/address/logic/commands/CommandTestUtil.java
#	src/test/java/seedu/address/storage/XmlSerializableConciergeTest.java

resolved.

teowz46 committed:
test: change try catch to throws Exception for tests where no exceptions are expected.

To standardise with other tests.

Commit:
Concierge: add guest list for checked-in guests and support for said list, ListCommand: refactor and update to accept new flag for listing checked-in guests, UiManager, LogicManager, and several Ui components: add support to update GuestListPanel to switch between displaying checked-in guests and archived guests

Commit:
ParserUtil: renamed FLAG_ARCHIVED_GUEST to FLAG_CHECKED_IN_GUEST and refactored occurences throughout

Commit:
XmlSerializableConcierge: add support to build checked-in guest list when reading in rooms and rooms' bookings

Commit:
ListCommand: refactor arguments

pikulet committed:
PasswordHashList.java: add test unit

pikulet committed:
ParserUtil.java: Add support for whitespace before passwords

pikulet committed:
Fix codacy issues

pikulet committed:
Fix typo

pikulet committed:
LogInManager.java: Add signOut method

pikulet committed:
Fix style issues

pikulet committed:
Add LogOutCommand

pikulet committed:
Fix style issues

pikulet committed:
Fix checkstyle issues

adamwth committed:
Merge pull request #164 from pikulet/user-guide-update

UserGuide.adoc: Update add, clear and login commands

Commit:
TypicalConcierge: fix bug

Commit:
Room: add support for adding tags

Commit:
Concierge: add addTag method

Commit:
Model: add addTag method

Commit:
TypicalTags: add class for typical tags for both guests and rooms

teowz46 committed:
test: add more expenses tests, fix existing tests

More tests and test files have been added for further integration testing of the expenses related classes.

In addition, ServiceCommand has been rewritten due to the removal of some interface methods in Model.

pikulet committed:
LogInManagerTest: add isSignedIn test

Commit:
fix tests

Commit:
Resolve merge conflict from #171

Commit:
fix style

Commit:
.gitignore: add dynamic XML test files to gitignore

Commit:
build.gradle: configure gradle build to show exception stack trace

Commit:
CommandTestUtil: attempt to fix travis bug

pikulet committed:
Merge branch 'master' into sign-in-feature

Commit:
fix style

teowz46 committed:
test: add integration tests for Expenses related classes in other classes

Other classes that interact with Expenses related classes, such as Concierge and Room, have new tests to verify the integration. Additional tests were also added targeting Expenses related classes in the XML class tests.

teowz46 committed:
all: fix checkstyle errors

teowz46 committed:
Merge branch 'master' into service

teowz46 committed:
ServiceCommandTest: fix Codacy style error

Codacy wanted a comment inside an empty method body so a comment was added.

pikulet committed:
Add LogInManagerTest

pikulet committed:
Fix checkstyle errors

Commit:
AddCommand: refactor, update all AddCommand related tests

teowz46 committed:
test: add more tests based on Coveralls

Additional tests were added based on the Coveralls report. Covered command parser, menu, missing room exception in XmlSerializableConcierge.

Commit:
RoomUtil: add utility methods for room commands, update and fix tests

Commit:
fix style

teowz46 committed:
all: fix checkstyle errors

Checkstyle errors were fixed and review comments were incorporated.

adamwth committed:
[v1.3] List checked in guests (#178)

* BookingPeriod.java: update isValidBookingPeriod to test if start date is strictly before end date

* BookingTest and BookingPeriodTest: add

* BookingsTest: add, add compareTo method to Guest and update compareTo of BookingPeriod and Booking

* CheckinCommand: minor fix to check-in logic and refactor, Room and Capacity: refactor Capacity to be enum and delete SingleRoom, DoubleRoom, SuiteRoom. RoomCard and RoomDetailedCard: minor fix to FXML element naming and standardize description

* UniqueRoomList: refactor to enforce no missing rooms, other tests: update to reflect compulsory rooms in all concierge xml files

* CheckinCommand: change input format, add tokenizer for room prefix

* CheckInCommand: rename from CheckinCommand, .gitignore: update

* ConciergeBuilder: add withMenu

* ConciergeTest: update tests

* ConciergeTest, ModelManagerTest, CheckInCommandTest: add

* CheckinCommandTest: update

* test: update AddCommandSystemTest to actually add booking

* TypicalRooms: add support for preset rooms with bookings

* Test utils: add support for more reusable test cases in Typical<Class>.java, update CheckinCommandTest and CheckoutCommandTest to use added test utils

* XmlAdaptedBookingPeriodTest: add test

* XmlAdaptedBookingTest: add test

* XmlAdaptedRoomTest: add tests

* fix style

* XmlUtilTest and XmlSerializableConciergeTest: update with new format of valid concierge, refactor to remove SOME hard-coded test cases from AB4

* XmlSerializableConciergeTest: update

* TypicalConcerge: Refactor name of getTypicalConcierge -> getTypicalConciergeClean, and getTypicalConciergeWithRoomBookings -> getTypicalConcierge

* fix style

* XmlConciergeStorageTest: update tests and relevant xml files used for tests

* XmlSerializableConciergeTest: update tests and add relevant XML files for updated tests

* XmlUtilTest: update the format of how tests are conducted, RoomListPanelHandle: fix NodeNotFoundException bug by changing FXML id of RoomCard

* DarkTheme.css: update with roomCardPane

* fix codacy style

* Concierge.java: add support for checked-in guest list, Rooms and other classes: refactor

* TypicalConcierge & TypicalRooms: Refactor to move all typical room settings to TypicalConcierge

* Concierge: add guest list for checked-in guests and support for said list, ListCommand: refactor and update to accept new flag for listing checked-in guests, UiManager, LogicManager, and several Ui components: add support to update GuestListPanel to switch between displaying checked-in guests and archived guests

* ParserUtil: renamed FLAG_ARCHIVED_GUEST to FLAG_CHECKED_IN_GUEST and refactored occurences throughout

* XmlSerializableConcierge: add support to build checked-in guest list when reading in rooms and rooms' bookings

* ListCommand: refactor arguments

* TypicalConcierge: fix bug

* Room: add support for adding tags

* Concierge: add addTag method

* Model: add addTag method

* TypicalTags: add class for typical tags for both guests and rooms

* fix tests

* fix style

* .gitignore: add dynamic XML test files to gitignore

* build.gradle: configure gradle build to show exception stack trace

* CommandTestUtil: attempt to fix travis bug

* fix style

* AddCommand: refactor, update all AddCommand related tests

* RoomUtil: add utility methods for room commands, update and fix tests

* fix style

pikulet committed:
Merge remote-tracking branch 'upstream/master' into sign-in-feature

teowz46 committed:
test: fix Codacy errors

adamwth committed:
Merge pull request #168 from pikulet/sign-in-feature

Sign-in feature

teowz46 committed:
Merge branch 'master' into service

Commit:
Resolve merge conflict from #168

teowz46 committed:
Merge branch 'master' into service

teowz46 committed:
all: fix bugs due to merging

Code could not compile after merging. Everything has been fixed and tests all pass.

teowz46 committed:
all: fix checkstyle errors due to merge

adamwth committed:
Merge pull request #158 from teowz46/service

ServiceCommand

adamwth committed:
[v1.3] Update test descriptions (#179)

* CheckinCommandParserTest: add test descriptions

* CheckoutCommandParserTest: add test descriptions

* fix style

adamwth committed:
[v1.3] Update UserGuide (#181)

* README.adoc: update app name to Concierge, fix travis url link

* DeveloperGuide: add sequence diagram for CheckinCommand

* README: update with AppVeyor badge

* BookingPeriod.java: update isValidBookingPeriod to test if start date is strictly before end date

* BookingTest and BookingPeriodTest: add

* BookingsTest: add, add compareTo method to Guest and update compareTo of BookingPeriod and Booking

* CheckinCommand: minor fix to check-in logic and refactor, Room and Capacity: refactor Capacity to be enum and delete SingleRoom, DoubleRoom, SuiteRoom. RoomCard and RoomDetailedCard: minor fix to FXML element naming and standardize description

* UniqueRoomList: refactor to enforce no missing rooms, other tests: update to reflect compulsory rooms in all concierge xml files

* CheckinCommand: change input format, add tokenizer for room prefix

* CheckInCommand: rename from CheckinCommand, .gitignore: update

* ConciergeBuilder: add withMenu

* ConciergeTest: update tests

* ConciergeTest, ModelManagerTest, CheckInCommandTest: add

* CheckinCommandTest: update

* test: update AddCommandSystemTest to actually add booking

* TypicalRooms: add support for preset rooms with bookings

* Test utils: add support for more reusable test cases in Typical<Class>.java, update CheckinCommandTest and CheckoutCommandTest to use added test utils

* XmlAdaptedBookingPeriodTest: add test

* XmlAdaptedBookingTest: add test

* XmlAdaptedRoomTest: add tests

* fix style

* XmlUtilTest and XmlSerializableConciergeTest: update with new format of valid concierge, refactor to remove SOME hard-coded test cases from AB4

* XmlSerializableConciergeTest: update

* TypicalConcerge: Refactor name of getTypicalConcierge -> getTypicalConciergeClean, and getTypicalConciergeWithRoomBookings -> getTypicalConcierge

* fix style

* XmlConciergeStorageTest: update tests and relevant xml files used for tests

* XmlSerializableConciergeTest: update tests and add relevant XML files for updated tests

* XmlUtilTest: update the format of how tests are conducted, RoomListPanelHandle: fix NodeNotFoundException bug by changing FXML id of RoomCard

* DarkTheme.css: update with roomCardPane

* README: add Codacy badge

* UserGuide: update check-in and check-out

* fix codacy style

* Concierge.java: add support for checked-in guest list, Rooms and other classes: refactor

* TypicalConcierge & TypicalRooms: Refactor to move all typical room settings to TypicalConcierge

* Concierge: add guest list for checked-in guests and support for said list, ListCommand: refactor and update to accept new flag for listing checked-in guests, UiManager, LogicManager, and several Ui components: add support to update GuestListPanel to switch between displaying checked-in guests and archived guests

* ParserUtil: renamed FLAG_ARCHIVED_GUEST to FLAG_CHECKED_IN_GUEST and refactored occurences throughout

* XmlSerializableConcierge: add support to build checked-in guest list when reading in rooms and rooms' bookings

* ListCommand: refactor arguments

* TypicalConcierge: fix bug

* Room: add support for adding tags

* Concierge: add addTag method

* Model: add addTag method

* TypicalTags: add class for typical tags for both guests and rooms

* fix tests

* fix style

* .gitignore: add dynamic XML test files to gitignore

* build.gradle: configure gradle build to show exception stack trace

* CommandTestUtil: attempt to fix travis bug

* fix style

* AddCommand: refactor, update all AddCommand related tests

* RoomUtil: add utility methods for room commands, update and fix tests

* fix style

* update Ui.png

* UserGuide: update

* fix style

teowz46 committed:
UserGuide: add section for service

The current service section is outdated. The section has now been updated with relevant instructions on the current implementation of ServiceCommand.

teowz46 committed:
Merge remote-tracking branch 'upstream/master' into service

JiaqingTan committed:
Modify "Find" Function (#169)

* Modify find function include both guests and rooms searching

* Change test details, convert prefixes over to new ones.

* Minor format change

* More minor change

* Change for Travis

* Files not in sync, have to commit and check travis again and again

* Update for Travis/Testing

* Replace code to utilize tokenizer

* Replace code to utilize tokenizer

* Made room list appear first, create tests for ListCommand and its parser

* More fixes

* AdocUpdates and Ui.png

* More fixes kill me soon

Commit:
Concierge: urgent fix to checkout

teowz46 committed:
Merge pull request #183 from teowz46/service

UserGuide: Service

Commit:
fix style

teowz46 committed:
UserGuide: remove "Coming in v1.3"

Because it has come.

teowz46 committed:
Merge remote-tracking branch 'upstream/master' into service

pikulet committed:
User guide: Documentation for add, clear, login/ logout, and select (#182)

* Update documentation for add, login, logout

* Add images to userguide

* Update select documentation

* Defensively change 1.3 features to 1.4

* Merge remote-tracking branch 'upstream/master' into user-guide-login

* ParserUtilTest.java: remove "address"

* Update README

* Fix README display image

adamwth committed:
Merge pull request #185 from adamwth/list-checked-in-guests

[v1.4] Update concierge checkout command

teowz46 committed:
UserGuide: add additional stuff for service command

Additional stuff missed out in the previous PR.

teowz46 committed:
Merge branch 'master' into service

teowz46 committed:
Minor edits for UserGuide (#190)

* UserGuide: remove "Coming in v1.3"

Because it has come.

* UserGuide: add additional stuff for service command

Additional stuff missed out in the previous PR.

adamwth committed:
[v1.4] User Guide update and Ui.png fix (#188)

* UserGuide: update checkout

* fix style

pikulet committed:
UserGuide.adoc: Update to provide admin account (#191)

* UserGuide.adoc: Update to provide admin account

* Add guide on adding new passwords

teowz46 committed:
Merge remote-tracking branch 'upstream/master' into service

# Conflicts:
#	docs/UserGuide.adoc

teowz46 committed:
Merge branch 'service' of https://github.com/teowz46/main into service

Commit:
UserGuide: update

Commit:
mainapp: update

Commit:
include passwords.json

Commit:
update preferences.json

Commit:
UserGuide: update

Commit:
fix style'

adamwth committed:
UserGuide: update (#195)

* UserGuide: update

* mainapp: update

* include passwords.json

* update preferences.json

* UserGuide: update

* fix style'

liliwei25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -311,15 +311,148 @@ image::UndoRedoActivityDiagram.png[width="650"]
+as parameters for the `AddCommand` constructor.
+
+An example of its usage:
+* `add n/Madith p/83141592 e/madith@themyth.com a/Recluse Street -r 041 from/
```
the address field is not necessary anymore right?

teowz46 committed:
Merge remote-tracking branch 'upstream/master'

liliwei25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -328,16 +461,71 @@ The list function now requires a flag after the 'list' command. Below are the tw
+=== Find Feature
+The Find feature expands upon the originally implemented FindCommand, allowing for the searching of both rooms and guests, with several filters.
+
+The UML Diagram for the current implementation of FindCommand is as follows:
```
I don't think the UML diagram was attached

liliwei25 commented on src/main/java/seedu/address/commons/events/ui/RoomListChangedEvent.java:
```diff
@@ -0,0 +1,26 @@
+import seedu.address.model.room.Room;
+
+/**
+ * Represents a room list change in the left UI column
```
Both also left UI column?

liliwei25 commented on src/main/java/seedu/address/logic/CommandHistory.java:
```diff
@@ -41,7 +41,9 @@ public void add(String userInput) {
         System.out.println(inputString);
-        CommandArchive.stringToFile(inputString);
+        // Fix this!!!
+        // CommandArchive.stringToFile(inputString);
```
Should exclude commented code

liliwei25 commented on src/main/java/seedu/address/logic/parser/CliSyntax.java:
```diff
@@ -9,11 +9,21 @@
+    public static final Prefix FLAG_ROOM_NO_BOOKINGS = new Prefix("-nb");
+    public static final Prefix FLAG_CHECKED_IN_GUEST = new Prefix("-cg");
+
+    public static final Prefix PREFIX_USERNAME = new Prefix("user/");
```
should group all the "PREFIX_XXXX" together

liliwei25 commented on src/main/java/seedu/address/logic/parser/FindCommandParser.java:
```diff
@@ -1,33 +1,203 @@
+        }
+
+        if (ParserUtil.arePrefixesPresent(argMultimap, FLAG_ROOM_HAS_BOOKINGS)
+                || ParserUtil.arePrefixesPresent(argMultimap, FLAG_ROOM_NO_BOOKINGS)) {
```
is it possible to reduce the nesting here?

liliwei25 commented on src/main/java/seedu/address/logic/parser/PasswordHashUtil.java:
```diff
@@ -0,0 +1,52 @@
+     * @return The SHA-256 hash of the supplied password
+     * @throws HashingException
+     */
+    public static String hash(String pw) throws HashingException {
```
should spell password in full for the parameter

liliwei25 commented on src/main/java/seedu/address/model/Concierge.java:
```diff
@@ -0,0 +1,342 @@
+    /**
+     * Checks out a room's first booking using its room number and remove the guest from the checked-in guest list
+     */
+    public void checkoutRoom(RoomNumber roomNumber) {
```
these two functions has a lot of repeated code. is it possible to call one function with the other instead?

liliwei25 commented on src/test/java/seedu/address/ui/MainWindowGuiChangeTest.java:
```diff
@@ -0,0 +1,61 @@
+        assertMainWindowDisplaysRoomList(mainWindow);
+
+        /*
+        @todo - Without Platform run later it encounters: Not on FX application thread; Even with it now, ui elements
```
remove unused code

liliwei25 commented on src/test/java/seedu/address/testutil/TypicalConcierge.java:
```diff
@@ -0,0 +1,83 @@
+     * ensure the model's versionedConcierge has an empty initial state list for all the tests to work.
+     */
+    public static Concierge getTypicalConciergeClean() {
+        Concierge ab = new Concierge();
```
maybe can use another name other than ab?

liliwei25 commented on src/test/java/seedu/address/model/login/PasswordHashListTest.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    // TODO: Add support to assert the success of parsing
+    /*
```
remove unused code

liliwei25 reviewed (commented):
* For CheckinCommandSequenceDiagram:
  * arrow for parse() should be below execute(), i.e. should follow the sequence of events
  * RoomNumber and CheckinCommand is cascaded in the same line
  * should the return arrow for result go back to LogicManager?
* For ListCommandUml:
  * should it still be AddressBookParser?
  * there should be one more flag right?
* There are two password.json files?

Good job! 👍 Good luck for your PE! 🙂

teowz46 committed:
Room: fix addExpense bug

Currently, there is no exception checking for adding expenses at the room level. This means that rooms that are not checked in or have no bookings are able to have expenses added to it. This is because the only checks currently reside at the Concierge level. In addition, there are no checks in the XML conversion for cases where expenses appear at rooms that are not checked in or have no bookings.

The lack of checks will lead to mysterious behaviour, such as a room without guests having expenses (which is in fact what the current sample Concierge generates). Users may also be able to edit the XML files to give expenses to rooms with no guests.

These bugs have been fixed by moving the checks from Concierge.java's addExpense method to Room.java's addExpense method. For the XML conversion, additional checks have been added for XmlAdaptedRoom.java. The corresponding tests have also been updated.

teowz46 committed:
all: fix checkstyle errors and failing test

Unused imports were removed and a failing test due to the newly introduced constraints on the storage of expenses in a room in XML was fixed.

adamwth committed:
Merge pull request #212 from teowz46/menu

Bug fix: Empty rooms can have Expenses

teowz46 committed:
RoomDetailedCard: add FlowPane for Expenses

Currently, the expenses of a room can only be seen as a total cost.

This makes the storage of the details of the individual expenses pointless.

A new FlowPane has been added to RoomDetailedCard, which displays the entire expenditure incurred by the guests of the current booking.

teowz46 committed:
Merge remote-tracking branch 'upstream/master' into expense

teowz46 committed:
Expenses: fix checkstyle error

teowz46 committed:
test: fix failing tests due to expenses

teowz46 committed:
RoomDetailedCardTest: add comparisons for "Expenses", "Active bookings", "All other bookings"

The current implementation that asserts two RoomDetailedCards are equal only check the room number, capacity, and tags.

"Expenses", "Active bookings" and "All other bookings" were not checked and this could lead to false positive assertions.

Checks for these areas have been implemented.

teowz46 committed:
RoomDetailedCardTest: add one more test for expenses FlowPane

An additional test was added to ensure that the FlowPane displaying the expenses shows the actual expenses of the room correctly.

adamwth committed:
[v1.4] Select command fix (#211)

* UserGuide: update

* mainapp: update

* include passwords.json

* update preferences.json

* UserGuide: update

* fix style'

* SelectCommand: select now only accepts valid indexes of the current displayed list

* .gitignore: update

teowz46 committed:
Expenses: fix toStringTotalCost method throwing MoneyLimitExceededException

It was reported that entering a cost that is too high for service will cause the RoomDetailedPanel to disappear. This was discovered to be due to MoneyLimitExceededException, which was thrown in the method toStringTotalCost in Expenses, which naively used a Money object to store the total cost, even though it is possible for the total cost to exceed the limit of values that a Money object can store. The method has been fixed to use BigDecimal to avoid this problem, assuming that BigDecimal can have virtually unlimited length.

teowz46 committed:
RoomDetailedCardHandle: modify equals()

Changed equals to compare every field.

adamwth committed:
Merge pull request #213 from teowz46/expense

Expenses detailed display on RoomDetailedPanel

Commit:
resolve merge conflict

adamwth committed:
[v1.4] User Guide update (#220)

* UserGuide: update

* fix style

* UserGuide: update checkout

* fix style

* UserGuide: update

* mainapp: update

* include passwords.json

* update preferences.json

* UserGuide: update

* fix style'

* UserGuide: update checkin, checkout, add

* UserGuide: update edit command

* fix style

adamwth committed:
[v1.4] Select command tests (#219)

* UserGuide: update

* mainapp: update

* include passwords.json

* update preferences.json

* UserGuide: update

* fix style'

* SelectCommand: select now only accepts valid indexes of the current displayed list

* .gitignore: update

* preferences.json: round off dimensions

* SelectCommandTest: add tests for checked-in guest list and room list, TypicalIndexes: rename

* TypicalConcierge: update javadoc, SelectCommandSystemTest: update javadoc, need to add new tests after filter checked-in guest functionality is up

* UserGuide: update find command to make description clearer

* SystemCommandSystemTest: add test for room select, other system tests: add methods to support testing of display of rooms and checked-in guests

* fix codacy errors

* XmlUtil: add method to support saving file with non existent file path by creating file path if missing

* add Deselect events, fix style

* UiManager, MainWindow, DetailedPanels: implement deselection feature; Checkout, Undo, Redo Commands: implement deselection of guest list

* listing commands: update with setDisplayedListFlag

* fix style

* XmlUtil: update javadoc

* remove DeselectRoomListEvent

Commit:
Merge remote-tracking branch 'upstream/master'

neilish3re committed:
Autocomplete feature (#221)

* Autocomplete feature

* remove unused prefixes from AutoCompleteManager class

* Autocomplete feature tests

adamwth committed:
[v1.4] Reassign command (#222)

* UserGuide: update

* mainapp: update

* include passwords.json

* update preferences.json

* UserGuide: update

* fix style'

* Concierge: add reassign method

* CheckoutCommand: refactor to use start date instead of booking period as identifier, update relevant tests

* fix tests

* gitignore: add preferences.json back

* Concierge: update reassign method

* ReassignCommand: add command and parser, along with methods to support it in relevant classes

* fix style

* ReassignCommandTest: add tests

* fix style

* fix style

* ReassignCommandParserTest: add and fix tests

* fix style

* RoomTest: add test for hasBookings and refactor

* ConciergeTest: add tests for checkin, checkout

* ConciergeTest: add more tests

* fix style

* ParserUtil: update parseDate

* Concierge: update reassign

* ConciergeTest: update tests

adamwth committed:
[v1.4] Docs update (#225)

* UserGuide: update

* mainapp: update

* include passwords.json

* update preferences.json

* UserGuide: update

* fix style'

* UserGuide: add reassign, terminology section

* DeveloperGuide: update checkin/checkout, CheckInCommandSequenceDiagram: fix

* DeveloperGuide: minor fix

* UserGuide: update reassign

Commit:
Merge remote-tracking branch 'upstream/master'

adamwth committed:
Revert "[v1.4] Docs update (#225)"

This reverts commit 909c735fdbd8866e5a57fd9cb3428024899fcb86.

adamwth committed:
Revert "[v1.4] Reassign command (#222)"

This reverts commit f59c3b72589b029fc3adaac023d56ab5f6c4584b.

adamwth committed:
Revert "Autocomplete feature (#221)"

This reverts commit bc79ebad2e101f34166fc0530fb49d8e76bf7bf0.

adamwth committed:
Merge pull request #227 from CS2103-AY1819S1-F11-2/revert-221-Autocomplete

Revert "Autocomplete feature"

Commit:
Merge remote-tracking branch 'upstream/master'

pikulet committed:
Create PPP template

JiaqingTan committed:
Bug Fixes, Changes and Added Functionality (#224)

* Modify find function include both guests and rooms searching

* Change test details, convert prefixes over to new ones.

* Minor format change

* More minor change

* Change for Travis

* Files not in sync, have to commit and check travis again and again

* Update for Travis/Testing

* Replace code to utilize tokenizer

* Replace code to utilize tokenizer

* Made room list appear first, create tests for ListCommand and its parser

* More fixes

* AdocUpdates and Ui.png

* More fixes kill me soon

* - Fix PE Dry Run Issues
- Change Find Command predicate argument types
- Change Find command to find both archieved and checked-in
- Feature to find room by guest name
- Change listCommandUml
- Add findCommandUml
- Portfolio Documentation

* Fixes to merge conflicts

* Fix for Not On FX Application Errors

* Even more unused import changes

* A bit more fixes and coveralls

* Fix for Integer.parseInt for capacity

* omg travis stop

pikulet committed:
Address the issues in the PE-dry-run (#229)

* MainWindow.java: Add functionality for help command to reset to new help page every time

* UserGuide.adoc: Explain help command

* Remove passwords.json and preferences.json

* AddCommand: disable adding of an outdated booking

pikulet committed:
Update README.adoc

- Update version number
- Rectify crediting of AB4

pikulet committed:
Remove edit command (#233)

* Remove edit command and Model::setGuest

* Update documentation to remove edit

pikulet committed:
[LogIn module improvements] Add default login account, erase command history with logout, add login requirement for checkin and checkout (#226)

* Add default login account

* Add support for saving json password file

* Fix checkstyle errors

* Update model\login testing units to match modified PasswordHashList

* UserGuide.adoc: Update login

* Model.java: Add functionality to reset versioned history upon logout

* passwords.json: Update default passwords file

* Remove passwords.json and preferences.json

These are causing the travis CI to fail

* Add Login testing units

* ClearCommandSystemTest.java: Add test for clear after logout

* Move addCommand utility from GuestUtil to BookingUtil

* CheckoutCommand.java: Add requirement for login

* CheckoutCommandSystemTest.java: Add system test and login check

* UserGuide.adoc: update checkout and logout

* Fix checkstyle errors

* CheckInCommand: mandate login

* CheckOutSystemTest.java: Add test cases

* CheckInCommandSystemTest.java: Add system test and login checks

* AddCommand.java: Add login checks

* Add LogIn for service and reassign

* Update UserGuide.adoc

* LogInCommand: update message usage

* VersionedConcierge.java: Undo removing of "final" qualifier

* Fix checkstyle errors

pikulet committed:
Merge remote-tracking branch 'upstream/master' into portfolio

pikulet committed:
Update UserGuide.adoc: Login

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Readme: update

Commit:
MainWindow and UiManager: change AB icon to Concierge icon

adamwth committed:
Merge pull request #234 from pikulet/portfolio

PPP: Restructure DeveloperGuide, update UserGuide: Login

Commit:
UserGuide: beautify

Commit:
UserGuide: update Ui.png and FAQ

Commit:
DeveloperGuide: update

Commit:
TypicalConcierge: change ab to concierge

Commit:
DeveloperGuide: add activity diagram for checkIn

teowz46 committed:
XmlAdaptedRoom: fix IllegalValueException due to expenses present in a checked-in but expired booking

The current implementation of XmlAdaptedRoom will only allow expenses to be added to rooms that have active and checked-in bookings.

However, this does not reflect the real-life scenario. For example, the user might exit the app on the same day as the last day of a booking with expenses, without checking out the room. The next day, the booking will become expired, and starting the app will thus throw an error. This behaviour should be prevented as it is possible that users may forget to check out after the day.

The section on adding expenses has been adjusted to allow for this. Expenses can now be added as long as there is a booking, and that booking is checked-in.

Commit:
adamwth.adoc: add PPP

Commit:
fix style

teowz46 committed:
SampleDataUtil: add two more expense types into sample

Two expense types were added for the purpose of representing discounts and typo corrections. This is necessary since there is no command for editing or deleting an expense, hence any adjustment will have to be added as an expense with either positive or negative value. These two expense types are thus general expense types for such a purpose.

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Resolve merge conflict

teowz46 committed:
UserGuide: update service section with the two adjustment expense types

A description for the usage of the two adjustment expense types has been added.

Commit:
UserGuide: add v2.0 section

Commit:
UserGuide: update

teowz46 committed:
UserGuide: fix minor stuff

Commit:
UserGuide: try fixing icon size

Commit:
AboutUs: update with link to PPP

Commit:
adamwth.adoc: update PPP

Commit:
UGDG: update

Commit:
UG: update image size

Commit:
UG: update image size

adamwth committed:
Merge pull request #238 from adamwth/adam-ppp

[v1.4] Adam's PPP

adamwth committed:
UserGuide: minor fix (#239)

JiaqingTan committed:
Update config.json

Fixed typo in my github user :(

neilish3re committed:
update author names for reposense (#230)

teowz46 committed:
DeveloperGuide: add ServiceCommand stuff with a flowchart

Stuff was added. May add more stuff.

teowz46 committed:
Merge remote-tracking branch 'upstream/master' into user-dev-guide

# Conflicts:
#	docs/UserGuide.adoc

adamwth committed:
[v1.4] Update Adam's PPP (#244)

* UserGuide: minor fix

* adamwth.adoc: update PPP

* adamwth.adoc: minor fix

* UserGuide: minor fix

* adamwth.adoc: update PPP

* UserGuide: update proportion of icon to text

teowz46 committed:
ModelClassDiagram: update with current structure

The ModelClassDiagram illustrated the Address Book structure. The diagram has been updated to reflect our Concierge structure.

teowz46 committed:
ModelClassDiagram: update the better diagram also

The other diagram was updated with our current terminology. UniqueRoomList and Menu were not included, otherwise it would be too cluttered (and also because I'm lazy).

pikulet committed:
UserGuide and DevGuide documentation (#235)

* Remove .PNG pictures, add .png pictures

* Dev Guide: Update add command

* UserGuide: update "add new account" section

* UserGuide: add note on parsing of passwords.json file

* update add activity diagram with bigger words

* DevGuide: login/logout current implementation

* Update dev guide: design considerations

* Update PPP with PRs reviewed

* AboutUs.adoc: Fix spelling of my name

* DevGuide: Add instructions for manual testing

* Update PPP

* DevGuide: Add LogInCommand sequence diagram

* Remove johndoe's PPP

* Remove unnecessary pictures in UserGuide

* Update Storage Diagram

* Add XML diagrams, update PPP

pikulet committed:
Fix implementation of resetUndoRedoHistory (#236)

* Rename password reference

* Fix style

* VersionedConcierge.java: Fix bug in resetUndoRedoHistory method

* Fix checkstyle errors

teowz46 committed:
DeveloperGuide: add sequence diagram for addExpense

Sequence diagram for the entire process of adding an expense is added. This is to supplement the flowchart with more detail on how an expense is added.

teowz46 committed:
DeveloperGuide: make addExpense sequence diagram bigger

The words are a bit unreadable.

neilish3re committed:
Autocomplete Feature (#237)

* Autocomplete Feature

* codacy touchups for the autocomplete feature tests

* Minor rearranging of import

* Change tag position in add command autocomplete

* Refactor autocomplete classes

* Update Method description for isCtrlDoubleTap()

teowz46 committed:
DeveloperGuide: add more design considerations

teowz46 committed:
all: fix checkstyle errors

pikulet committed:
Documentation updates, test cases, comments (#246)

* Model/login: Update method comments

* PasswordHashListTest.java: Abstract String constants

* SampleDataUtil: Rename method to getDefaultPasswordHashList

Reduce confusion with PasswordHashListTest.getSamplePasswordHashList

* AddCommandTest.java: Add Booking tests

* Refactor tests for model/login

* AddCommandSystemTest: Add room/booking tests

* DevGuide: Remove "get started on programming" section

* Fix phrasing and style

* Added links in PPP

* MainApp.java: Fix version number

* PPP: Fix typo

* Update DeveloperGuide

JiaqingTan committed:
Change Application Name to "Concierge", edit PPP (#243)

* Modify find function include both guests and rooms searching

* Change test details, convert prefixes over to new ones.

* Minor format change

* More minor change

* Change for Travis

* Files not in sync, have to commit and check travis again and again

* Update for Travis/Testing

* Replace code to utilize tokenizer

* Replace code to utilize tokenizer

* Made room list appear first, create tests for ListCommand and its parser

* More fixes

* AdocUpdates and Ui.png

* More fixes kill me soon

* - Fix PE Dry Run Issues
- Change Find Command predicate argument types
- Change Find command to find both archieved and checked-in
- Feature to find room by guest name
- Change listCommandUml
- Add findCommandUml
- Portfolio Documentation

* Fixes to merge conflicts

* Fix for Not On FX Application Errors

* Even more unused import changes

* A bit more fixes and coveralls

* Fix for Integer.parseInt for capacity

* omg travis stop

* Update PPP, change gitignore to include config.json to change app name to "Concierge"

* Try fix no newline eof

* Update gitignore to include config.json again

* Edits to devguide and PPP

neilish3re committed:
UserGuide, DevGuide, PPP documentation (#245)

* UserGuide and DevGuide documentation (Neil)

* Add PPP documentation and minor additions to UserGuide and DevGuide

teowz46 committed:
docs: add my own PPP

teowz46 committed:
XmlAdaptedRoomTest: add more tests to increase coverage

Tests targeted at some of the highlighted parts in Coveralls. Not sure why the other highlighted parts were not covered.

teowz46 committed:
XmlAdaptedRoomTest: fix checkstyle errors

neilish3re committed:
Update doc images from jpg to png format (#249)

* Update doc images from jpg to png format

* Remove old jpg images

teowz46 committed:
UserGuide: add minor corrections

teowz46 committed:
UG and DG updates for Expenses stuff (#247)

* XmlAdaptedRoom: fix IllegalValueException due to expenses present in a checked-in but expired booking

The current implementation of XmlAdaptedRoom will only allow expenses to be added to rooms that have active and checked-in bookings.

However, this does not reflect the real-life scenario. For example, the user might exit the app on the same day as the last day of a booking with expenses, without checking out the room. The next day, the booking will become expired, and starting the app will thus throw an error. This behaviour should be prevented as it is possible that users may forget to check out after the day.

The section on adding expenses has been adjusted to allow for this. Expenses can now be added as long as there is a booking, and that booking is checked-in.

* SampleDataUtil: add two more expense types into sample

Two expense types were added for the purpose of representing discounts and typo corrections. This is necessary since there is no command for editing or deleting an expense, hence any adjustment will have to be added as an expense with either positive or negative value. These two expense types are thus general expense types for such a purpose.

* UserGuide: update service section with the two adjustment expense types

A description for the usage of the two adjustment expense types has been added.

* UserGuide: fix minor stuff

* DeveloperGuide: add ServiceCommand stuff with a flowchart

Stuff was added. May add more stuff.

* ModelClassDiagram: update with current structure

The ModelClassDiagram illustrated the Address Book structure. The diagram has been updated to reflect our Concierge structure.

* ModelClassDiagram: update the better diagram also

The other diagram was updated with our current terminology. UniqueRoomList and Menu were not included, otherwise it would be too cluttered (and also because I'm lazy).

* DeveloperGuide: add sequence diagram for addExpense

Sequence diagram for the entire process of adding an expense is added. This is to supplement the flowchart with more detail on how an expense is added.

* DeveloperGuide: make addExpense sequence diagram bigger

The words are a bit unreadable.

* DeveloperGuide: add more design considerations

* all: fix checkstyle errors

* docs: add my own PPP

* XmlAdaptedRoomTest: add more tests to increase coverage

Tests targeted at some of the highlighted parts in Coveralls. Not sure why the other highlighted parts were not covered.

* XmlAdaptedRoomTest: fix checkstyle errors

* UserGuide: add minor corrections

neilish3re committed:
Change uppercase PNG to lowercase png (#250)

teowz46 committed:
Merge remote-tracking branch 'upstream/master' into user-dev-guide

teowz46 committed:
UserGuide and DeveloperGuide: minor edits from previous PR

teowz46 committed:
UserGuide: use important icon

teowz46 committed:
UserGuide: fix some formatting errors

teowz46 committed:
UG and DG updates (#252)

* XmlAdaptedRoom: fix IllegalValueException due to expenses present in a checked-in but expired booking

The current implementation of XmlAdaptedRoom will only allow expenses to be added to rooms that have active and checked-in bookings.

However, this does not reflect the real-life scenario. For example, the user might exit the app on the same day as the last day of a booking with expenses, without checking out the room. The next day, the booking will become expired, and starting the app will thus throw an error. This behaviour should be prevented as it is possible that users may forget to check out after the day.

The section on adding expenses has been adjusted to allow for this. Expenses can now be added as long as there is a booking, and that booking is checked-in.

* SampleDataUtil: add two more expense types into sample

Two expense types were added for the purpose of representing discounts and typo corrections. This is necessary since there is no command for editing or deleting an expense, hence any adjustment will have to be added as an expense with either positive or negative value. These two expense types are thus general expense types for such a purpose.

* UserGuide: update service section with the two adjustment expense types

A description for the usage of the two adjustment expense types has been added.

* UserGuide: fix minor stuff

* DeveloperGuide: add ServiceCommand stuff with a flowchart

Stuff was added. May add more stuff.

* ModelClassDiagram: update with current structure

The ModelClassDiagram illustrated the Address Book structure. The diagram has been updated to reflect our Concierge structure.

* ModelClassDiagram: update the better diagram also

The other diagram was updated with our current terminology. UniqueRoomList and Menu were not included, otherwise it would be too cluttered (and also because I'm lazy).

* DeveloperGuide: add sequence diagram for addExpense

Sequence diagram for the entire process of adding an expense is added. This is to supplement the flowchart with more detail on how an expense is added.

* DeveloperGuide: make addExpense sequence diagram bigger

The words are a bit unreadable.

* DeveloperGuide: add more design considerations

* all: fix checkstyle errors

* docs: add my own PPP

* XmlAdaptedRoomTest: add more tests to increase coverage

Tests targeted at some of the highlighted parts in Coveralls. Not sure why the other highlighted parts were not covered.

* XmlAdaptedRoomTest: fix checkstyle errors

* UserGuide: add minor corrections

* UserGuide and DeveloperGuide: minor edits from previous PR

* UserGuide: use important icon

* UserGuide: fix some formatting errors

teowz46 committed:
UserGuide: rearrange stuff in service

teowz46 committed:
Merge remote-tracking branch 'upstream/master' into user-dev-guide

# Conflicts:
#	docs/UserGuide.adoc

teowz46 committed:
More UG updates (#253)

* XmlAdaptedRoom: fix IllegalValueException due to expenses present in a checked-in but expired booking

The current implementation of XmlAdaptedRoom will only allow expenses to be added to rooms that have active and checked-in bookings.

However, this does not reflect the real-life scenario. For example, the user might exit the app on the same day as the last day of a booking with expenses, without checking out the room. The next day, the booking will become expired, and starting the app will thus throw an error. This behaviour should be prevented as it is possible that users may forget to check out after the day.

The section on adding expenses has been adjusted to allow for this. Expenses can now be added as long as there is a booking, and that booking is checked-in.

* SampleDataUtil: add two more expense types into sample

Two expense types were added for the purpose of representing discounts and typo corrections. This is necessary since there is no command for editing or deleting an expense, hence any adjustment will have to be added as an expense with either positive or negative value. These two expense types are thus general expense types for such a purpose.

* UserGuide: update service section with the two adjustment expense types

A description for the usage of the two adjustment expense types has been added.

* UserGuide: fix minor stuff

* DeveloperGuide: add ServiceCommand stuff with a flowchart

Stuff was added. May add more stuff.

* ModelClassDiagram: update with current structure

The ModelClassDiagram illustrated the Address Book structure. The diagram has been updated to reflect our Concierge structure.

* ModelClassDiagram: update the better diagram also

The other diagram was updated with our current terminology. UniqueRoomList and Menu were not included, otherwise it would be too cluttered (and also because I'm lazy).

* DeveloperGuide: add sequence diagram for addExpense

Sequence diagram for the entire process of adding an expense is added. This is to supplement the flowchart with more detail on how an expense is added.

* DeveloperGuide: make addExpense sequence diagram bigger

The words are a bit unreadable.

* DeveloperGuide: add more design considerations

* all: fix checkstyle errors

* docs: add my own PPP

* XmlAdaptedRoomTest: add more tests to increase coverage

Tests targeted at some of the highlighted parts in Coveralls. Not sure why the other highlighted parts were not covered.

* XmlAdaptedRoomTest: fix checkstyle errors

* UserGuide: add minor corrections

* UserGuide and DeveloperGuide: minor edits from previous PR

* UserGuide: use important icon

* UserGuide: fix some formatting errors

* UserGuide: rearrange stuff in service

teowz46 committed:
AboutUs: update my PPP link

neilish3re committed:
Updated LogicComponentClassDiagram (#251)

teowz46 committed:
AboutUs: update my pic

teowz46 committed:
Merge remote-tracking branch 'upstream/master' into user-dev-guide

teowz46 committed:
Merge pull request #254 from teowz46/user-dev-guide

AboutUs update

neilish3re committed:
Renaming of parser in Logic Component Class Diagram (#255)

teowz46 committed:
AddExpenseSequenceDiagram: fix out of alignment objects (#256)

teowz46 committed:
README: update link for ConciergeFinal.png to show properly in index.html and github (#259)

pikulet committed:
Remove gitter badge

adamwth committed:
[v1.4] Docs update (#257)

* UG: update checkin/out, ContactUs: update email

* PPP: update

* asciidoctor: remove line 381 - fixes html links appearing in pdf

* DG: add update UML diagrams

* fix style

* update name as shown in IVLE

* PPP: add terminology

JiaqingTan committed:
Update to UserGuide for List + Find (#261)

* Modify find function include both guests and rooms searching

* Change test details, convert prefixes over to new ones.

* Minor format change

* More minor change

* Change for Travis

* Files not in sync, have to commit and check travis again and again

* Update for Travis/Testing

* Replace code to utilize tokenizer

* Replace code to utilize tokenizer

* Made room list appear first, create tests for ListCommand and its parser

* More fixes

* AdocUpdates and Ui.png

* More fixes kill me soon

* - Fix PE Dry Run Issues
- Change Find Command predicate argument types
- Change Find command to find both archieved and checked-in
- Feature to find room by guest name
- Change listCommandUml
- Add findCommandUml
- Portfolio Documentation

* Fixes to merge conflicts

* Fix for Not On FX Application Errors

* Even more unused import changes

* A bit more fixes and coveralls

* Fix for Integer.parseInt for capacity

* omg travis stop

* Update PPP, change gitignore to include config.json to change app name to "Concierge"

* Try fix no newline eof

* Update gitignore to include config.json again

* Edits to devguide and PPP

* Update UserGuide List and Find feature

pikulet committed:
Update documentation (#260)

* Update diagrams

* UserGuide: update undoable commands

* Update grammar

adamwth committed:
[v1.4] SampleDataUtil: add yesterday-today booking to r/005 (#263)

* SampleDataUtil: add yesterday-today booking to r/005

* SampleDataUtil: add expired booking to r/006

pikulet committed:
Update PPP links (#262) - forced merge

* Update PPP links

teowz46 committed:
README (#264)

* README: update link for ConciergeFinal.png to show properly in index.html and github

* README: show Ui.png

* README: minor edit

teowz46 committed:
DeveloperGuide: add manual testing instructions for service (#265)

pikulet committed:
Add instructions for manual testing: list, find, autocomplete (#266)

* Update PPP links

* Update PPP links

* Update PPP links

* Add instructions for manual testing: autocomplete, find, list

* Change room number for list command

adamwth committed:
DG: update manual test instructions for checkin, checkout, reassign (#267) - forced merge

* SampleDataUtil: add yesterday-today booking to r/005

* SampleDataUtil: add expired booking to r/006

* readme fix

* DG: update

* undo readme change

adamwth committed:
DG fix (#268)

* DG: fix description
